README.md

```diff
@@ -90,54 +106,151 @@ on: issue_comment
-        uses: platisd/clang-tidy-pr-comments@master
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          clang_tidy_fixes: fixes.yaml
-          pull_request_id: ${{ env.pr_id }}
+    - name: Download analysis results
+      uses: actions/github-script@v3.1.0
+      with:
+        script: |
+          let artifacts = await github.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: ${{github.event.workflow_run.id }},
+          });
+          let matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "clang-tidy-result"
+          })[0];
+          let download = await github.actions.downloadArtifact({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              artifact_id: matchArtifact.id,
+              archive_format: "zip",
+          });
+          let fs = require("fs");
+          fs.writeFileSync("${{github.workspace}}/clang-tidy-result.zip", Buffer.from(download.data));
+    - name: Set environment variables
+      run: |
+        mkdir clang-tidy-result
+        unzip clang-tidy-result.zip -d clang-tidy-result
+        echo "pr_id=$(cat clang-tidy-result/pr-id.txt)" >> $GITHUB_ENV
+        echo "pr_head_repo=$(cat clang-tidy-result/pr-head-repo.txt)" >> $GITHUB_ENV
+        echo "pr_head_ref=$(cat clang-tidy-result/pr-head-ref.txt)" >> $GITHUB_ENV
+    - uses: actions/checkout@v2
+      with:
+        repository: ${{ env.pr_head_repo }}
+        ref: ${{ env.pr_head_ref }}
+        persist-credentials: false
+    - name: Redownload analysis results
+      uses: actions/github-script@v3.1.0
+      with:
+        script: |
+          let artifacts = await github.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: ${{github.event.workflow_run.id }},
+          });
+          let matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "clang-tidy-result"
+          })[0];
+          let download = await github.actions.downloadArtifact({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              artifact_id: matchArtifact.id,
+              archive_format: "zip",
+          });
+          let fs = require("fs");
+          fs.writeFileSync("${{github.workspace}}/clang-tidy-result.zip", Buffer.from(download.data));
+    - name: Extract analysis results
+      run: |
+        mkdir clang-tidy-result
+        unzip clang-tidy-result.zip -d clang-tidy-result
+    - name: Run clang-tidy-pr-comments action
+      if: ${{ hashFiles( 'clang-tidy-result/fixes.yml' ) != '' }}
+      uses: platisd/clang-tidy-pr-comments@master
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        clang_tidy_fixes: clang-tidy-result/fixes.yml
+        pull_request_id: ${{ env.pr_id }}
```

Comment:
This is getting very complex! :scream: Isn't it much easier to use [actions/download-artifact](https://github.com/actions/download-artifact)?

Reply:
> Isn't it much easier to use actions/download-artifact?

It is used here:

```
let download = await github.actions.downloadArtifact({
```

:) But it's not so trivial to download artifact **from another workflow**. `actions/download-artifact` is intended to be used to share artifacts **between different jobs in the same workflow**. If you want to grab an artifact from a different workflow, things are getting complicated.

Reply:
Aha I see! Very useful piece of code then regardless of this Action! :grin: 
I now have also realized what this workflow _actually_ does. :stuck_out_tongue: 

OK, I like it but README is getting very extensive. Can you add some Markdown (sub)headers for these cases and create a table of contents right before the `What` section for easier navigation?

Reply:
> OK, I like it but README is getting very extensive. Can you add some Markdown (sub)headers for these cases and create a table of contents right before the What section for easier navigation?

Yes, of course. I'll try to sort it out :)